include/openthread-types.h

```diff
@@ -37,13 +37,19 @@
 
 #include <stdint.h>
 #include <stdbool.h>
+
 #include <platform/toolchain.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
+ * This type represents the OpenThread instance structure.
+ */
+typedef struct otInstance otInstance;
```

Comment:
We should avoid having clients of OpenThread needing to pull in definitions for internal structures.  Can we just typedef a `void` instead?